src/exo/LoopIR_compiler.py

```diff
@@ -375,11 +375,10 @@ def from_lines(x):
     public_fwd_decls = []
 
     # Body contents
-    memory_code = _compile_memories(find_all_mems(proc_list))
```

Comment:
I presume you made this change so that if a builtin is onlu used within the definition of an instruction, then we don't need to generate its "globals". I think I can agree with that, however, this might cause problems for memories e.g.

@instr(...)
def foo(...):
     a: var @ SOME_NEW_MEMORY
     # Do something with `a`

Now, to be fair, I don't know of any case where you really need to do this, but it legal Exo code and we should support it.

Reply:
This change was made because we have to run precision analysis on the proc_list before calling `_compile_builtins`. But yeah, it might be confusing not to generate globls for instructions. I'll change that.